### PR TITLE
Update ImageSharp to use Primitives from SixLabors.Core

### PR DESCRIPTION
**Pull request by tocsoft**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/JimBobSquarePants/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practise as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
This removes the structs from ImageSharp dealing with Point, Size Rectangle etc and replaces them with equivalents from SixLabors.Core.

### CODE REVIEW

**Comment by tocsoft** [comment]
should be prettty straight forward, just worth a bit of a sanity check cause its changing a lot of places

File: Scratch/Program.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+using System.Numerics;
+using ImageSharp;
+
+namespace Scratch
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            using (var image = Image.Load(@"C:\Users\tocso\Desktop\92d8758c-471c-11e7-81d8-62ec301b092a (1).jpg")) {
```
Comment (JimBobSquarePants) [inline Scratch/Program.cs]: You might wanna change this.

Comment (tocsoft) [inline Scratch/Program.cs]: woops that whole file/project shouldn't be in here will be removed.

File: src/ImageSharp.Drawing/Paths/FillRectangle.cs [hunk 2]
```diff
@@ -23,10 +24,10 @@ public static partial class ImageExtensions
         /// <param name="shape">The shape.</param>
         /// <param name="options">The options.</param>
         /// <returns>The <see cref="Image{TPixel}"/>.</returns>
-        public static Image<TPixel> Fill<TPixel>(this Image<TPixel> source, IBrush<TPixel> brush, Rectangle shape, GraphicsOptions options)
+        public static Image<TPixel> Fill<TPixel>(this Image<TPixel> source, IBrush<TPixel> brush, RectangleF shape, GraphicsOptions options)
             where TPixel : struct, IPixel<TPixel>
         {
-            return source.Fill(brush, new SixLabors.Shapes.Rectangle(shape.X, shape.Y, shape.Width, shape.Height), options);
+            return source.Fill(brush, new SixLabors.Shapes.RectangularePolygon(shape.X, shape.Y, shape.Width, shape.Height), options);
```
Comment (tocsoft) [inline src/ImageSharp.Drawing/Paths/FillRectangle.cs]: probably not, probably wasn't thinking to much about it... that can be addressed on the Shapes repo.

File: src/ImageSharp.Drawing/Paths/ShapeRegion.cs [hunk 3]
```diff
@@ -42,12 +45,12 @@ public ShapeRegion(IPath shape)
         /// <inheritdoc/>
         public override int Scan(float y, Span<float> buffer)
         {
-            Vector2 start = new Vector2(this.Bounds.Left - 1, y);
-            Vector2 end = new Vector2(this.Bounds.Right + 1, y);
-            Vector2[] innerbuffer = ArrayPool<Vector2>.Shared.Rent(buffer.Length);
+            PointF start = new PointF(this.Bounds.Left - 1, y);
+            PointF end = new PointF(this.Bounds.Right + 1, y);
+            PointF[] innerbuffer = ArrayPool<PointF>.Shared.Rent(buffer.Length);
```
Comment (tocsoft) [inline src/ImageSharp.Drawing/Paths/ShapeRegion.cs]: yeah your right... i'll move it over to using Buffer<>

File: src/ImageSharp.Drawing/Paths/FillRectangle.cs (hunk 2 above)
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Paths/FillRectangle.cs]: Cool. Was just curious, Need to get stuck into that source. 👍

**Comment by tocsoft** [comment]
@JimBobSquarePants we good to go on this now? or do @dlemstra and @antonfirsov want to take a look over it first?

**Comment by JimBobSquarePants** [comment]
Yeah, it's a simple enough change. I'll merge it.

**Pull request merged**